File: src/main/scala/uk/co/ridentbyte/view/util/ColumnConstraintsBuilder.scala
package uk.co.ridentbyte.view.util

import javafx.scene.layout.{ColumnConstraints, Priority}

case class ColumnConstraintsBuilder() {

  val columnConstraints = new ColumnConstraints

  def withHgrow(priority: Priority): ColumnConstraintsBuilder = {
    columnConstraints.setHgrow(priority)
    this
  }

  def withPercentageWidth(percentage: Int): ColumnConstraintsBuilder = {
    columnConstraints.setPercentWidth(percentage)
    this
  }

  def build: ColumnConstraints = columnConstraints

}

File: src/main/scala/uk/co/ridentbyte/view/util/ColumnConstraintsBuilder.java
package uk.co.ridentbyte.view.util;

import javafx.scene.layout.ColumnConstraints;
import javafx.scene.layout.Priority;

public class ColumnConstraintsBuilder {

    private ColumnConstraints columnConstraints;

    public ColumnConstraintsBuilder() {
        this.columnConstraints = new ColumnConstraints();
    }

    public ColumnConstraintsBuilder withHgrow(Priority priority) {
        columnConstraints.setHgrow(priority);
        return this;
    }

    public ColumnConstraintsBuilder withPercentageWidth(int percentage) {
        columnConstraints.setPercentWidth(percentage);
        return this;
    }

    public ColumnConstraints build() {
        return columnConstraints;
    }
}
